src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,226 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+export type PaginationProps = PropsWithChildren<{
+  className?: string;
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  // Optional props
+  showIcon?: boolean;
+  showIconOnly?: boolean;
+  showOnlyPreviousAndNext?: boolean;
+  tableDataPagination?: boolean;
```

Comment:
If you adopt my suggestion above, this will not be required anymore.

Reply:
done